### PR TITLE
Override ERB::Util#html_escape with Module#prepend

### PR DESCRIPTION
**Pull request by k0kubun**
### Motivation / Background

This Pull Request has been created because, with Ruby 3.2 or the upcoming release of erb.gem, `ERB::Util#html_escape` will be faster than `CGI.escapeHTML` in the case that it escapes nothing because `ERB::Util#html_escape` will not duplicate the argument, unlike `CGI.escapeHTML`. https://bugs.ruby-lang.org/issues/19102

Calling the original `ERB::Util#html_escape` with `super` is preferable not only for the above performance reason but also for making it more permissive about other monkey patches using `Module#prepend`. I tried to use `Module#prepend` there myself, but I gave it up because it wasn't compatible with the monkey patch of Rails not using `Module#prepend`.

### Detail

This Pull Request changes `ERB::Util#unwrapped_html_escape` to use `super` to call the original `ERB::Util#html_escape`.

### Additional information

The first definition of `html_escape` was to call the original `ERB::Util#html_escape` with `super`, and then it's renamed to `unwrapped_html_escape` with `alias`. It's just for calling the original method with `super`, and the actual `html_escape` is implemented in the second definition.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [x] CI is passing.

### CODE REVIEW

File: activesupport/lib/active_support/core_ext/erb/util.rb [hunk 1]
```diff
@@ -1,6 +1,31 @@
 # frozen_string_literal: true
 
 class ERB
+  module ActiveSupportUtil
```
Comment (casperisfine) [inline activesupport/lib/active_support/core_ext/erb/util.rb:4]: I think that module should be `ActiveSupport::CoreExt::ERBUtil`

**Comment by casperisfine** [comment]
Other than that LGTM, happy to merge it.

File: activesupport/lib/active_support/core_ext/erb/util.rb [hunk 2]
```diff
@@ -17,34 +45,8 @@ module Util
...
-      unwrapped_html_escape(s).html_safe
-    end
-
-    silence_redefinition_of_method :h
-    alias h html_escape
-
-    module_function :h
-
-    singleton_class.silence_redefinition_of_method :html_escape
-    module_function :html_escape
```
Comment (casperisfine) [inline activesupport/lib/active_support/core_ext/erb/util.rb:35]: Hum, `module_function` makes the instance method private right?

We might be missing a couple `private` calls to be 100% compatible.

File: activesupport/lib/active_support/core_ext/erb/util.rb (hunk 1 above)
Comment (k0kubun) [reply activesupport/lib/active_support/core_ext/erb/util.rb]: Fixed it and pushed https://github.com/rails/rails/pull/46573/commits/0fc2c6f2100cc91560dc3262b64172ac62f866e3.

File: activesupport/lib/active_support/core_ext/erb/util.rb (hunk 2 above)
Comment (k0kubun) [reply activesupport/lib/active_support/core_ext/erb/util.rb:35]: Made those methods private, and pushed https://github.com/rails/rails/pull/46573/commits/cedf12d579fbe441edbad33e3f94aba7a0762ddb

Comment (k0kubun) [reply activesupport/lib/active_support/core_ext/erb/util.rb:35]: :thinking: Looks like I need to figure out something for making it `private` correctly

Comment (k0kubun) [reply activesupport/lib/active_support/core_ext/erb/util.rb:35]: https://github.com/rails/rails/pull/46573/commits/d6c42b3fba456d1882682a5348610309164b1edd

**Pull request merged**